test.js

```diff
@@ -39,22 +39,62 @@ var validURLs = [
   'http://host.xz/path/to/repo.git/',
   'https://host.xz/path/to/repo.git/',
   'git@github.com:user/some-project.git',
-  'git@github.com:user/some_project.git'
+  'git@github.com:user/some_project.git',
+  'user@host.xz:/path/to/repo.git/',
+  'user@host.xz:~user/path/to/repo.git/',
+  'user@host.xz:path/to/repo.git',
+  'user@host.xz:path/to/repo.git/',
+  'user123@host.xz:path/to/repo.git/',
+  'User.123-1@host.xz:path/to/repo.git/',
+  'git://github.com/ember-cli/ember-cli#v0.1.0',
+  'git://github.com/ember-cli/ember-cli#ff786f9f',
+  'git://github.com/ember-cli/ember-cli#master',
+  'git://github.com/ember-cli/ember-cli#gh-pages',
+  'git://github.com/ember-cli/ember-cli#quick_fix',
+  'git://github.com/ember-cli/ember-cli#Quick-Fix',
+  'git@github.com:user/project',
+  'git@github.com:user/some-project',
+  'git@github.com:user/some_project',
+  'https://github.com/user/project',
+  'http://github.com/user/project',
+  'git@192.168.101.127:user/project',
+  'https://192.168.101.127/user/project',
+  'http://192.168.101.127/user/project',
+  'ssh://user@host.xz:port/path/to/repo/',
+  'ssh://user@host.xz/path/to/repo/',
+  'ssh://host.xz:port/path/to/repo/',
+  'ssh://host.xz/path/to/repo/',
+  'ssh://user@host.xz/path/to/repo/',
+  'ssh://host.xz/path/to/repo/',
+  'ssh://user@host.xz/~user/path/to/repo/',
+  'ssh://host.xz/~user/path/to/repo/',
+  'ssh://user@host.xz/~/path/to/repo',
+  'ssh://host.xz/~/path/to/repo',
+  'git://host.xz/path/to/repo/',
+  'git://host.xz/~user/path/to/repo/',
+  'http://host.xz/path/to/repo/',
+  'https://host.xz/path/to/repo/',
+  'git@github.com:user/some-project',
+  'git@github.com:user/some_project',
+  'user@host.xz:/path/to/repo/',
+  'user@host.xz:~user/path/to/repo/',
+  'user@host.xz:path/to/repo',
+  'user@host.xz:path/to/repo/',
+  'user123@host.xz:path/to/repo/',
+  'User.123-1@host.xz:path/to/repo/',
 ];
 
 var invalidURLs = [
-  'git@github.com:user/some_project.gitfoo',
-  'git@github.com:user/some_project.git/foo',
```

Comment:
not sure why these were removed. if the tests fail on these then it would be a regression